src/Layout.ts

```diff
@@ -1438,9 +1438,9 @@ export class Union extends Layout<LayoutObject> {
   configGetSourceVariant: (getSourceVariant: (src: LayoutObject) => VariantLayout | undefined) => void;
 
   constructor(
-      discr: Layout<LayoutObject> | UnionDiscriminator,
-      defaultLayout: Layout<LayoutObject> | null,
-      property: string
+      discr: UInt | UIntBE | ExternalLayout | UnionDiscriminator,
+      defaultLayout?: Layout<LayoutObject> | null,
```

Comment:
The JSDoc says this is nullable but not optional. The code coerces `undefined` to `null`. Which position are we going to take: the docs rule or the code rules?

```suggestion
      defaultLayout: Layout<LayoutObject> | null,
```

Reply:
I guess the code is an actual state of being, and docs are always can be stale.

Reply:
To proof my words openbook (aka serum) use union with an optional arguments in mind